packages/wonder-blocks-modal/src/components/flexible-panel.tsx

```diff
@@ -0,0 +1,211 @@
+import * as React from "react";
+import {PropsFor, View} from "@khanacademy/wonder-blocks-core";
+import type {StyleType} from "@khanacademy/wonder-blocks-core";
+
+import {focusStyles} from "@khanacademy/wonder-blocks-styles";
+import {StyleSheet} from "aphrodite";
+import {semanticColor} from "@khanacademy/wonder-blocks-tokens";
+import ModalContent from "./modal-content";
+import CloseButton from "./close-button";
+import theme from "../theme";
+
+export type BackgroundStyles = {
+    /**
+     * The background color of the panel. Defaults to semanticColor.surface.primary.
+     */
+    backgroundColor?: string;
+    /**
+     * The background image URL or gradient.
+     */
+    backgroundImage?: string;
+    /**
+     * How the background image should be repeated.
+     */
+    backgroundRepeat?: "repeat" | "no-repeat" | "repeat-x" | "repeat-y";
+    /**
+     * How the background image should be positioned.
+     */
+    backgroundPosition?: string;
+    /**
+     * How the background image should be sized.
+     */
+    backgroundSize?: string;
+    /**
+     * How the background image should fit within its container.
+     */
+    objectFit?: "fill" | "contain" | "cover" | "none" | "scale-down";
+};
+
+type RenderProps = {
+    title: React.ReactNode | string;
+};
+
+type Props = {
+    /**
+     * The main heading of the FlexiblePanel. Used to label the dialog.
+     */
+    title?: React.ReactNode | string;
+    /**
+     * Optional id reference for the main heading to label the parent dialog via aria-labelledby.
+     */
+    titleId?: string;
+    /**
+     * The main contents of the FlexiblePanel. All other parts of the panel
+     * are positioned around it.
+     */
+    content:
+        | React.ReactElement<PropsFor<typeof ModalContent>>
+        | ((slots: RenderProps) => React.ReactNode)
+        | React.ReactNode;
+    /**
+     * The optional background styles for the panel.
+     * If not provided, defaults to semanticColor.surface.primary background color.
+     */
+    backgroundStyles?: BackgroundStyles;
```

Comment:
For components that have multiple elements within it that can be styled, we've been using a `styles` prop with different keys for the elements that can get styled! 

Would it make sense here to combine the `style` and `backgroundStyles` props into: 
```
styles?: {
  panel?: StyleType;
  background?: StyleType;
}
```

I see though that `style` and `backgroundStyles` are applied to the same element though! So I wonder if only support a `style` prop would suffice?

Reply:
I think we need `styles` to support the background and `closeButton`!